### PR TITLE
Fix watch channel with nil seek position

### PR DESCRIPTION
**Pull request by sunby**
issue: #11649
Signed-off-by: sunby <bingyi.sun@zilliz.com>

### CODE REVIEW

File: internal/datacoord/services.go [hunk 1]
```diff
@@ -115,12 +115,7 @@ func (s *Server) AssignSegmentID(ctx context.Context, req *datapb.AssignSegmentI
 			zap.Uint32("count", r.GetCount()))
 
 		if coll := s.meta.GetCollection(r.CollectionID); coll == nil {
```
Comment (congqixia) [inline internal/datacoord/services.go:117]: Shall we give up query root when collection is not found here?

Comment (sunby) [reply internal/datacoord/services.go:117]: > Shall we give up query root when collection is not found here?

oh I should call s.GetCollection()

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**